From 5e2eb18b457ab536b63a55cef7b0b323c65a1f91 Mon Sep 17 00:00:00 2001
From: Jari Nippula <jari.nippula@intel.com>
Date: Fri, 17 Feb 2017 10:43:12 +0200
Subject: [PATCH 2/2] Type-c in DEVICE mode if not connection

Set Type-c port to DEVICE mode by default in case there is no
cable connected to the port. This keeps dwc3 module sane in case it
is waken up from suspend while no cable connection to Host PC.

Signed-off-by: Jari Nippula <jari.nippula@intel.com>
---
 drivers/usb/typec/typec_wcove.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/typec/typec_wcove.c b/drivers/usb/typec/typec_wcove.c
index c30bb36..fa241f7 100644
--- a/drivers/usb/typec/typec_wcove.c
+++ b/drivers/usb/typec/typec_wcove.c
@@ -414,9 +414,9 @@ static irqreturn_t  wcove_typec_irq(int irq, void *data)
 
 		wcove_typec_func(wcove, WCOVE_FUNC_ORIENTATION,
 				 WCOVE_ORIENTATION_NORMAL);
-		/* Host mode by default */
+		/* Device mode by default */
 		wcove_access_xhci(wcove, true);
-		wcove_typec_func(wcove, WCOVE_FUNC_ROLE, WCOVE_ROLE_HOST);
+		wcove_typec_func(wcove, WCOVE_FUNC_ROLE, WCOVE_ROLE_DEVICE);
 		wcove_access_xhci(wcove, false);
 		/* reset the pd sink state */
 		if (wcove->pd_port_num >= 0)
-- 
2.7.4

